go/tasks/plugins/array/k8s/task.go

```diff
@@ -60,33 +81,36 @@ func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeCl
 	if t.Config.RemoteClusterConfig.Enabled {
 		podTemplate.OwnerReferences = nil
 	}
-	var args []string
-	if len(podTemplate.Spec.Containers) > 0 {
-		args = append(podTemplate.Spec.Containers[0].Command, podTemplate.Spec.Containers[0].Args...)
-		podTemplate.Spec.Containers[0].Command = []string{}
-	} else {
+	if len(podTemplate.Spec.Containers) == 0 {
 		return LaunchError, errors2.Wrapf(ErrReplaceCmdTemplate, err, "No containers found in podSpec.")
 	}
+	containerIndex, err := getTaskContainerIndex(&podTemplate)
+	if err != nil {
+		return LaunchError, err
+	}
+
+	args := append(podTemplate.Spec.Containers[containerIndex].Command, podTemplate.Spec.Containers[containerIndex].Args...)
+	podTemplate.Spec.Containers[containerIndex].Command = []string{}
 
 	indexStr := strconv.Itoa(t.ChildIdx)
 	podName := formatSubTaskName(ctx, tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), indexStr)
 
 	pod := podTemplate.DeepCopy()
 	pod.Name = podName
-	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, corev1.EnvVar{
+	pod.Spec.Containers[containerIndex].Env = append(pod.Spec.Containers[containerIndex].Env, corev1.EnvVar{
 		Name:  FlyteK8sArrayIndexVarName,
 		Value: indexStr,
 	})
 
-	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, arrayJobEnvVars...)
+	pod.Spec.Containers[containerIndex].Env = append(pod.Spec.Containers[containerIndex].Env, arrayJobEnvVars...)
 	taskTemplate, err := tCtx.TaskReader().Read(ctx)
 	if err != nil {
 		return LaunchError, errors2.Wrapf(ErrGetTaskTypeVersion, err, "Unable to read task template")
 	} else if taskTemplate == nil {
 		return LaunchError, errors2.Wrapf(ErrGetTaskTypeVersion, err, "Missing task template")
 	}
 	inputReader := array.GetInputReader(tCtx, taskTemplate)
-	pod.Spec.Containers[0].Args, err = template.Render(ctx, args,
+	pod.Spec.Containers[containerIndex].Args, err = template.Render(ctx, args,
```

Comment:
I guess this should apply to cmd as well?

Reply:
we already combine the command and args: https://github.com/flyteorg/flyteplugins/pull/182/files#diff-01ee392f44bbd91e1e5f4afd999327e02f8f1df54fd5b5b576478e903cbcb3c2R92

(for the record, primary container always has an sdk supplied command + args)

Reply:
I don't think, generally speaking, we should make assumptions based on how we implemented Flytekit... the underlying spec allows cmd & args and I don't think this plugin should change that behavior... What do yo think?

Reply:
Also, I thought we are getting rid of the args/cmd manipulation altogether from this function, right?

Reply:
oh derp, yes. thanks & done